tests/ut/runtime/test_runtime_base.py

```
@@ -273,6 +273,134 @@ def test_require_ready_consulted_by_alloc_tensor():
+    tensor = DeviceTensor(0x2000, (4,), torch.float32, buffer=buffer)
+
+    probe._require_owned_resident_tensor(tensor, "Argument")
+    assert buffers.scans == 0
+
+
+def test_ownership_check_rejects_a_freed_handle():
+    buffer = _OwnedBuffer(0x2000, owner_worker_id=5)
+    probe = _OwnershipProbe(_ScanCountingBuffers())  # nothing live
+    tensor = DeviceTensor(0x2000, (4,), torch.float32, buffer=buffer)
+
+    with pytest.raises(ValueError, match="not a live allocation.*on worker_id=5"):
```

Ruff (0.16.2)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)